tests/test_coiled.py

```diff
@@ -0,0 +1,37 @@
+import os
+import uuid
+
+import coiled
+import dask.dataframe as dd
+import pandas
+from dask.distributed import Client
+
+SOFTWARE = os.environ["SOFTWARE_ENV"]
+
+
+def test_quickstart():
+    cluster = coiled.Cluster(
+        software=SOFTWARE,
+        name="nyc-quickstart_" + str(uuid.uuid4()),
+        account="dask-engineering",
+        n_workers=10,
+        backend_options={"spot": False},
+    )
```

Comment:
yup, I just saw the clusters weren't shutting down I cancel the workflows. 
I'm having 12 clusters coming up because of the way the matrix is set, I think most of this is redundant. Investigating.

Reply:
Restricting to only run CI on pushes to `main` by default should remove the redundant CI builds